debezium-core/src/main/java/io/debezium/pipeline/signal/channels/MemorySignalChannel.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.pipeline.signal.channels;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.config.CommonConnectorConfig;
+import io.debezium.pipeline.signal.SignalRecord;
+
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+/**
+ * The class responsible for processing of signals delivered to Debezium via a JVM memory. 
+ * The signal message must have the following structure:
+ * <ul>
+ * <li>{@code id STRING} - the unique identifier of the signal sent, usually UUID, can be used for deduplication</li>
+ * <li>{@code type STRING} - the unique logical name of the code executing the signal</li>
+ * <li>{@code data STRING} - the data in JSON format that are passed to the signal code
+ * </ul>
+ */
+public class MemorySignalChannel implements SignalChannelReader{
```

Comment:
@ismailsimsek One question - how do you envision usage of this channel? In DebeziumEngine?

Reply:
@jpechane I was thinking for Debezium Server, Using to trigger Read only Snapshot Signals 
something like:

1. User fills form and submits data from web UI of Debezium Server + Quarks 
2. Debezium Server Backend adds this to MemorySignalChannel
3. Debezium process processes it

what do you think? is it  reasonable feature, doable?

Reply:
I think for that we plan to use https://issues.redhat.com/browse/DBZ-6424
We intend to expose this via Jolokia so it will be available using REST endpoint.

Reply:
Thats great, then i will remove it.